editoast/src/main.rs

```diff
@@ -164,3 +175,172 @@ fn generate(
+        };
+
+        // WHEN
+        let result = import_railjson(args, pg_config, chartos_config);
+
+        // THEN
+        assert!(result.is_err())
+    }
+
+    #[test]
+    fn import_railjson_test_with_wrong_railjson_version() {
+        // GIVEN
+        let pg_config: PostgresConfig = Default::default();
+        let chartos_config = Default::default();
+
+        let railjson: RailJson = RailJson {
+            version: String::from("0"),
+            operational_points: vec![],
+            routes: vec![],
+            switch_types: vec![],
+            switches: vec![],
+            track_section_links: vec![],
+            track_sections: vec![],
+            speed_sections: vec![],
+            catenaries: vec![],
+            signals: vec![],
+            buffer_stops: vec![],
+            detectors: vec![],
+        };
+
+        let infra_name = generate_infra_name(&String::from("deleteme"));
+        let tmp_file = generate_railjson_temp_file(railjson);
+
+        let args: ImportRailjsonArgs = ImportRailjsonArgs {
+            infra_name,
+            railjson_path: tmp_file.path().into(),
+        };
+
+        // WHEN
+        let result = import_railjson(args, pg_config, chartos_config);
+
+        // THEN
+        assert!(result.is_err())
+    }
+
+    #[test]
+    fn import_railjson_test_ok() {
+        // GIVEN
+        let pg_config: PostgresConfig = Default::default();
+        let database_url = &pg_config.url();
+        let chartos_config = Default::default();
+
+        let railjson: RailJson = RailJson {
+            version: RAILJSON_VERSION.to_string(),
+            operational_points: vec![],
+            routes: vec![],
+            switch_types: vec![],
+            switches: vec![],
+            track_section_links: vec![],
+            track_sections: vec![],
+            speed_sections: vec![],
+            catenaries: vec![],
+            signals: vec![],
+            buffer_stops: vec![],
+            detectors: vec![],
+        };
+
+        let infra_name = generate_infra_name(&String::from("deleteme"));
+        let tmp_file = generate_railjson_temp_file(railjson.clone());
+
+        let args: ImportRailjsonArgs = ImportRailjsonArgs {
+            infra_name: infra_name.clone(),
+            railjson_path: tmp_file.path().into(),
+        };
+
+        // WHEN
+        let result = import_railjson(args, pg_config, chartos_config);
+
+        // THEN
+        assert!(matches!(result, Ok(_)));
+
+        let conn = PgConnection::establish(database_url).expect("Error while connecting DB");
+        let imported_infra = Infra::find_by_name(&conn, infra_name);
+        let infra: Infra = imported_infra.unwrap();
+        let infra_id = infra.id;
+        let railjson_version_created = infra.railjson_version;
+
+        assert_eq!(railjson_version_created, railjson.version);
+
+        // FINALLY
+        assert!(Infra::delete(infra_id, &conn).is_ok());
```

Comment:
The strategy to clear the database at the end of the test is not clean.
One way to do it is:

- Split `import_railjson` function in two function.
  - The first function parse the configuration to create the connection and opening the file (`conn` and `railjson_file`).
  - The second fuction takes a connection as argument, it can do the rest.

This allow to only call the second function in your test using a test transaction that will clear all modification afterward.

Reply:
My best guess is that this an integration test, not a unit test. Ok for splitting. What about keeping the semantic (with delete at the end) but moving it to test folder ?